### DISCUSSION TITLE
Vercel Deployment Error After Refactoring Code

### DISCUSSION THREAD

**Discussion by MeenaFoerderer** [neuefische/questions#352] [Web] [answered]
Hello,
I have completely refactored my code and implemented dynamic routing. Now I have a problem with the deployment at Vercel. It appears to have problems with an object key (`title`) from my dummy array and with the routing. Everything works fine with `npm run dev`. 
I have built in an early return to solve the problem, but it doesn't change. Also `title` appear to be and work correct everywhere. I also did a redeployment without cache, didn't help either.

`npm run build` throws me the following error:
<img width="936" alt="Bildschirmfoto 2022-11-30 um 16 38 04" src="https://user-images.githubusercontent.com/105375266/204841541-b76037e6-25e9-4a3c-b6ab-36a554d9cd7d.png">

These are the error messages at Vercel:
![Bildschirmfoto 2022-11-30 um 16 39 52](https://user-images.githubusercontent.com/105375266/204842100-f33d17ac-2f95-4ca3-82ab-a41f5ec464a9.png)
![Bildschirmfoto 2022-11-30 um 16 40 32](https://user-images.githubusercontent.com/105375266/204842227-dd7c9390-1a83-4c9a-af1c-7c99e5844a40.png)

This is my [pull request/repo](https://github.com/MeenaFoerderer/Capstone-project/pull/16)

**Comment by actyralf** [comment]
Hi Meena,

please check the files in the directories highlighted in the build report. (e.g. `/pages/[date]/[room]/talks/[id].js` . Next.js expects EVERY file in the `/pages` directory to export a valid React component as default export. E.g. if you comment out the content of some of these files because you don't need them anymore, Next.js will complain.

**Reply by MeenaFoerderer** [reply]
Thanks Ralf, 
in my case, this didn't cause a problem. The thing was that my early return was not at the right spot. It had to be moved up in the code, before a `.find` method where I destructured at the same time.

**Comment by MeenaFoerderer** [comment, marked as answer]
The early return was not at the right spot. The solution was to move it up in the code, before a .find method was implemented where I destructured at the same time.

It works like this now:
![Bildschirmfoto 2022-11-30 um 17 30 27](https://user-images.githubusercontent.com/105375266/204854325-7fde833d-ded5-42b2-8343-ef39279bf9a4.png)

Before that, the early return was below the destructuring part.
